src/Admin/AbstractAdmin.php

```diff
@@ -1379,6 +1382,12 @@ public function setTemplates(array $templates)
      */
     public function setTemplate($name, $template)
     {
+        @trigger_error(sprintf(
+            'Method "%s" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0. '
+            .'Use TemplateRegistry services instead.',
```

Comment:
```suggestion
            'Method "%s()" is deprecated since sonata-project/admin-bundle 3.x and will be removed in 4.0.'
            .' Use TemplateRegistry services instead.',
```